### PR TITLE
 games-simulation/openrct2: version bump

### PR DESCRIPTION
**Pull request by lorem-ipsum**
also mirror updates in nightly.

Patch disabling tests requiring game files is submitted upstream:
OpenRCT2/OpenRCT2#7853

### CODE REVIEW

**Comment by jansegre** [comment]
I noticed that this version requires doesn't build on `gcc<=6` (because of the use of string_view, and most importantly support for it was dropped). Even though gcc-7 is stable, I wonder if this should be encoded in the ebuild.

File: games-simulation/openrct2/openrct2-0.2.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+		-DWITH_TESTS="$(usex test)"
+		-DDOWNLOAD_TITLE_SEQUENCES=OFF
+		-DDOWNLOAD_OBJECTS=OFF
+		-DBUILD_SHARED_LIBS=ON
+	)
+	if use test ; then
+		mycmakeargs+=(
+			-DSYSTEM_GTEST=ON
+			-DDISABLE_RCT2_TESTS=ON
+			)
```
[the remark this answers is not in the document]
Comment (lorem-ipsum) [reply games-simulation/openrct2/openrct2-0.2.0.ebuild]: I changed it to be aligned with the variable as it was used for all others. Same for PATCHES.

File: games-simulation/openrct2/openrct2-0.2.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	inherit git-r3
+	SRC_URI=""
+else
+	KEYWORDS="~amd64 ~x86"
+	SRC_URI="https://github.com/OpenRCT2/OpenRCT2/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	S="${WORKDIR}/OpenRCT2-${PV}"
+fi
+
+TSV="0.1.2"
+SRC_URI+=" https://github.com/OpenRCT2/title-sequences/releases/download/v${TSV}/title-sequence-v${TSV}.zip -> ${PN}-title-sequence-v${TSV}.zip "
```
[the remark this answers is not in the document]
Comment (lorem-ipsum) [reply games-simulation/openrct2/openrct2-0.2.0.ebuild]: Yep, looks better. Still too long, but better.

File: games-simulation/openrct2/Manifest [hunk 3]
```diff
@@ -1,3 +1,5 @@
 DIST openrct2-0.1.2.tar.gz 6769785 BLAKE2B 0cc6ef6c68c1db6c822dd6038bc2d115f487a52b5891316273ee36db1713913baa2f119dabace88a37c16f87c90b14cf84df75a24b26b22571e98966afd2adbc SHA512 489dac96e1eda1449680f98a4b500eeb7b3f1ffbdaa5a5c701300071eb3a7862c65fa5a81e769083927b95a14914db54307724756a9841cff0c769c9e5471adc
```
[the remark this answers is not in the document]
Comment (lorem-ipsum) [reply games-simulation/openrct2/Manifest]: Yeah, didn't think about the special meaning of mirror in ebuild terms. Should be clearer now.

**Comment by lorem-ipsum** [comment]
@jansegre I don't know how to encode this. Building with clang works fine (at least for recent versions). I think it is tied to he newer profiles.

**Comment by mgorny** [comment]
@lorem-ipsum: grep for `tc-is-gcc`, you'd see a lotta ebuilds checking GCC versions ;-).

**Comment by lorem-ipsum** [comment]
I would leave it as is. According to [this](https://gcc.gnu.org/onlinedocs/libstdc++/manual/status.html) GCC 7.1 is required for proper support of `string_view`. Checking properly for clang is quite difficult as it depends on the standard library used. By default the GCC lib is used, also requiring GCC 7.1. However when sys-libs/libcxx is used it must be checked against its version (acording to (this)[https://libcxx.llvm.org/cxx1z_status.html] version 4.0 should be enough).
A proper check would require to compile a simple test file, checking for the feature macros listed in the linked GCC feature list. This appears to be overkill for a pre merge check as it should be done in configure anyways.

**Comment by mgorny** [comment]
Ya, if `configure` checks for it, that's fine.

**Pull request closed without merging**